src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -130,6 +141,34 @@ def __init__(self, msConfig, logger=None):
         self.wfCounters = {'cleaned': {},
                            'archived': {'normalArchived': 0,
                                         'forceArchived': 0}}
+        self.globalLocks = set()
+
+    def getGlobalLocks(self):
+        """
+        Fetches the list of 'globalLocks' from wmstats server and the list of
+        'parentLocks' from request manager. Stores/updates the unified set in
+        the 'globalLocks' instance variable. Returns the resultant unified set.
+        :return: A union set of the 'globalLocks' and the 'parentLocks' lists
+        """
+        globalLocks = set()
+        parentLocks = set()
+        self.logger.info("Fetching globalLocks list from wmstats server.")
+        try:
+            globalLocks.update(set(self.wmstatsSvc.getGlobalLocks()))
```

Comment:
I'm not sure there is any performance/memory difference, but we do not need to update a set, we only need to assign it. In other words, I'm more in favor of something like:
```
globalLocks = set(self.wmstatsSvc.getGlobalLocks())
```

and there is no need to initialize it in the beginning of the method.

Same comment goes to the `parentLocks` block below.

Reply:
Done